tpu/make-cluster-commands.sh

```diff
@@ -0,0 +1,58 @@
+#! /bin/bash
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+NAME=$1
+ZONE=$2
+NUM_WORKERS=$3
+shift
+
+# If there is a proxy, use it (For Googlers).
+if which corp-ssh-helper > /dev/null; then
+  EXTRA_SSH_ARGS="-o ProxyCommand=corp-ssh-helper %h %p"
+else
+  EXTRA_SSH_ARGS=
+fi
+
+cat > cluster-run.sh <<EOF
+PIDS=()
+mkdir -p /tmp/dtensor/pids
+for ((i=0;i<$NUM_WORKERS;i++)); do
+  echo Running on worker "\${i}" "\$*"
+  gcloud compute tpus tpu-vm ssh $NAME --zone=$ZONE --worker=\$i -- -t -q -n "-o ProxyCommand=corp-ssh-helper %h %p" bash -c -l "'\$*'" > /tmp/${NAME}_\${i}.log 2>&1 &
+  CPID=\$!
+  PIDS+=("\${CPID}")
+  echo \$i > "/tmp/dtensor/pids/\${CPID}"
+done
+
+while [[ -n "\${PIDS}" ]]; do
+  wait -p CPID -fn "\${PIDS[@]}"
+  PIDS=(\${PIDS[@]/\$CPID})
+  NODE=\$(cat /tmp/dtensor/pids/\${CPID})
+  echo ===Log from "\${NODE}"===
+  cat /tmp/${NAME}_\${NODE}.log
+done
+EOF
+
+cat > cluster-bcast.sh <<EOF
```

Comment:
Since this script just does the scp, how about call it cluster-scp?

Reply:
On the GPU side, we have a cluster-bcast.sh script so I kept the same name to be consistent.

Reply:
scp usually copies to a single destination host. bcast implies 1 to many.